qcodes/instrument/parameter.py

```diff
@@ -115,15 +115,23 @@ class _SetParamContext:
     >>> assert abs(dac.voltage() - v) <= tolerance
 
     """
-    def __init__(self, parameter):
+    def __init__(self, parameter: "Parameter", value: Any):
         self._parameter = parameter
+        self._value = value
         self._original_value = self._parameter._latest["value"]
 
+        self._value_is_changing = self._value != self._original_value
+
+        if self._original_value is None and self._value_is_changing:
+            self._original_value = self._parameter.get()  # type: ignore
```

Comment:
Following the merge of #1738 please don't use bare ignores. use `--show-error-codes` to get the error code needed and do `ignore[errorcode]`

Reply:
Gotcha!